packages/cypress-commands/src/setups/enableNetworkShim/captureRequests.js

```diff
@@ -166,8 +166,9 @@ function processDuplicatedRequest({
         // RequestStub was already nonDeterministic, responseBody is already an array
         const matchingResponseBodyIndex = isNotModified
             ? // When isNotModified the response body is empty and the browser is expected
-              // to return the last known response, so we do the same.
-              requestStub.responseBody.lenght - 1
+              // to return the last known response, so we do the same by returning the 
+              // value of the last responseLookup.
+              requestStub.responseLookup[requestStub.responseLookup.length - 1]
```

Comment:
This was wrong: For non-deterministic requests, the `responseBody` array is just a list of *unique* response bodies. The full sequence of requests, and which body should be returned when, is stored in the `responseLookup` array. So to get the last known response we should look at the value of the last item of the `responseLookup`.